strings/validator/validator.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package validator for validating strings.
+// The validators are used for instance in filter function.
+// A number of specific validators ready-to-use are also included.
+package validator
+
+import (
+	"net"
+	"regexp"
+	"strings"
+)
+
+// Validator interface must be implemented by all validators
+type Validator interface {
```

Comment:
Rather than a type, why not define this in terms of simple functions?

e.g. https://play.golang.org/p/XJG8QmDUEk5

Reply:
Looks better. I will change. Thanks for the comment. This is why I called on slack. I do not know all fine points of `go` yet and can take a wrong turn.

Reply:
I really like this pattern, with first-class functions as the crux.  Rob Pike wrote this up in https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html

Reply:
Me too. I did not know that you can use a `object.method()` in place of a function in `go`. I don't think there is something similar in Java or C++, probably why I did such an awkward first attempt.